### PR TITLE
Docs unreleased banner

### PR DESCRIPTION
**Pull request by tkknight**
### Description

For a better user experience it would be useful for the documentation to clearly state when you are viewing the **latest** unreleased docs (from master).

This change uses the `_templates/layout.html` to add extra content, in this case to the `document` block (the "body" content of every page on the docs).

The rendered documentation for this PR: https://tkknight-iris-test-doc.readthedocs.io/en/latest/ (may change until this PR is out of draft)

This is a part solution to #3992.

### Logic that triggers showing the notice 

When the docs are built on the Read the Docs service there is a shell environment variable named  `READTHEDOCS_VERSION` that should be set to: `latest` / `stable` / `v3.0.1`/ etc.  This variable is pulled into `conf.py` and then passed into the sphinx build via the `html_context variable`.

The banner/notice will **only** be shown when rendering on the read the docs service and when using the latest (against master) docs.

### Presentation style of the notice

The logic and the banner is in the `_templates/layout.html` file, using the `document` block (see https://www.sphinx-doc.org/en/master/templating.html).   

The `document` means the banner would be added at the top of the body of **every** page on the docs.  If the user scrolls down the banner will not be visible.

I have tried variations where the banner is fixed at the top of every page but my css skills are limited and it would render over the top of the existing content, meaning the iris logo was a little overwitten - so I aborted this approach.

#### TO DO

- [x] Choose (or another option) the presentation style of the notice.  
  - [x] Option 1: Red banner (needs a little bit of css to create).  **This is the one that will be used**
  - [x] Option 2: Use a read the docs theme note/warning.  This fits with the styliing, however it may not be as visible to users as a red banner.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/src/_templates/layout.html [hunk 1]
```diff
@@ -1,5 +1,36 @@
...
+{# This uses blocks.  See: 
+     https://www.sphinx-doc.org/en/master/templating.html
+#}
+
+/*---------------------------------------------------------------------------*/
+
+{%- block document %}
+    {% if READTHEDOCS and rtd_version == 'latest' %}
+        <div id="slim-red-box-message">
+            You a viewing the <b>latest</b> unreleased documentation
```
Comment (rcomer) [inline docs/src/_templates/layout.html:12]: You ***are*** viewing

File: docs/src/_templates/layout.html [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
+            <b>v{{ version }}</b>.  You may prefer a 
+            <a href="https://scitools-iris.readthedocs.io/en/stable/">stable</a>
+            version.
+        </div>
+        <p></p>
+        
+        <div class="admonition warning">
+            <p class="admonition-title">Warning</p>
+                You a viewing the <b>latest</b> unreleased documentation
```
Comment (rcomer) [inline docs/src/_templates/layout.html:21]: As above

**Review by rcomer — COMMENTED** [review]
I like the red.  Can't miss it!

**Comment by bjlittle** [comment]
@tkknight I've taken this out of draft... It's ready for review, right?

**Comment by tkknight** [comment]
> @tkknight I've taken this out of draft... It's ready for review, right?

Just removed the alternate banner.  Once the docs have finished building it will be ready for review.

**Comment by bjlittle** [comment]
Just ping me when it's good to go, and I'll get to it asap 👍

**Comment by tkknight** [comment]
> Just ping me when it's good to go, and I'll get to it asap 👍

Ping @bjlittle.  Took a little bit to pass the cirrus checks.  The link check got stuck.  I cancelled the task and rerun manually, passed ok.

File: docs/src/whatsnew/latest.rst [hunk 3]
```diff
@@ -91,6 +91,9 @@ This document explains the changes made to Iris for this release
 
 #. `@bjlittle`_ added the |PyPI|_ badge to the `README.md`_. (:pull:`4004`)
 
+#. `@tkknight`_ added a banner at the top of every page if the unreleased
+   development documentatiion is being viewed. (:pull:`3999`)
+
```
Comment (bjlittle) [inline docs/src/whatsnew/latest.rst:96]: @tkknight could you add clarification to the reader that this only applies to `rtd`... that's how I understand it, right?

Comment (rcomer) [reply docs/src/whatsnew/latest.rst:96]: Typo "if" --> "of".

**Comment by tkknight** [comment]
finally, maybe, possibly, good to go.

**Review by bjlittle — APPROVED** [review]
@tkknight LGTM :+1: 

Awesome, thanks!

**Pull request merged**